### PR TITLE
remove transactionCategory in favor of more types

### PR DESCRIPTION
**Pull request by brad-decker**
### Rationale
Our Transaction object has keys for both `type` and `transactionCategory` but the following are true:
1. Incoming transactions had no type, only a transactionCategory.
2. Cancel and retry transactions were special case `type`s.
3. All other transactions were `standard` `type` which just meant to look at the `transactionCategory`

This pull request attempts to simplify the process of getting a view of what a transaction is by eliminating the `transactionCategory` field in preference of the `type` field.

There is only one additional case that needs to be covered in the future possibly: Signature requests, which currently have no category or type.

Supersedes #9545

### CODE REVIEW

**Review by ryanml — DISMISSED** [review]
looks like a well applied sweeping update of categories -> types ++

**Comment by brad-decker** [comment]
@ryanml had to rebase, no changes were made :)

**Review by ryanml — DISMISSED** [review]

**Comment by brad-decker** [comment]
Waiting for other reviewers to surface until tomorrow, at which point I'll remove the do not merge label and merge.

File: app/scripts/controllers/transactions/index.js [hunk 1]
```diff
@@ -812,10 +809,27 @@ export default class TransactionController extends EventEmitter {
   }
 
   /**
-    Returns a "type" for a transaction out of the following list: simpleSend, tokenTransfer, tokenApprove,
-    contractDeployment, contractMethodCall
-  */
-  async _determineTransactionCategory(txParams) {
+   * @typedef { 'transfer' | 'approve' | 'transferfrom' | 'contractInteraction'| 'sentEther' } InferrableTransactionTypes
```
Comment (Gudahtt) [inline app/scripts/controllers/transactions/index.js:812]: Neat! This is appreciated.

File: test/unit/app/controllers/network/stubs.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import {
-  TRANSACTION_CATEGORIES,
   TRANSACTION_STATUSES,
   TRANSACTION_TYPES,
```
Comment (Gudahtt) [inline test/unit/app/controllers/network/stubs.js:3]: There are two references to the now-deleted `STANDARD` type in this file

**Review by Gudahtt — COMMENTED** [review]
Looks great! I found one minor issue with a stubs file, but everything looks good aside from that.

I found a few remaining references to a "transaction category" but they were in reference to the _transaction group category_, which is still a thing that exists. We might want to consider changing the nomenclature there at some point to avoid this confusion, but that can happen at another time.

**Review by rekmarks — COMMENTED** [review]
I didn't do a full review, so I won't throw in my full weight with an approval, but the changes make sense to me and the migration looks good 👍

File: test/unit/app/controllers/network/stubs.js (hunk 2 above)
Comment (brad-decker) [reply test/unit/app/controllers/network/stubs.js:3]: Fixed up, thanks for finding!

**Comment by brad-decker** [comment]
> I found a few remaining references to a "transaction category" but they were in reference to the transaction group category, which is still a thing that exists. We might want to consider changing the nomenclature there at some point to avoid this confusion, but that can happen at another time.

@Gudahtt 100% agree. I almost did that here but changed my mind haha. I will eliminate this as part of my proposal for regrouping transactions

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by Gudahtt** [comment]
I investigated the impact this might have on mobile, since we do sync some transactions.

They use our transactions here: https://github.com/MetaMask/metamask-mobile/blob/develop/app/core/Engine.js#L394

It looks like they don't use the `type` field or the `transactionCategory` field, so this change should have no impact.

**Pull request merged**